business/checkers/destinationrules/multi_match_checker.go

```diff
@@ -114,8 +114,8 @@ func checkCollisions(validations models.IstioValidations, destinationRulesName s
 func addError(validations models.IstioValidations, destinationRuleNames []string) models.IstioValidations {
 	for _, destinationRuleName := range destinationRuleNames {
 		key := models.IstioValidationKey{Name: destinationRuleName, ObjectType: DestinationRulesCheckerType}
-		checks := models.BuildCheck("More than one DestinationRules for the same host subset combination",
-			"warning", "spec/host")
+		checks := models.BuildWarningCheck("More than one DestinationRules for the same host subset combination",
```

Comment:
I don't fully understand this approach.
Moving from using a "warning" string to a custom method BuildWarningCheck doesn't make really a difference.
As @burmanm comments, if we have a generic Build() and we have a table configurable somewhere perhaps it may sense, but for this scenario I'd recommend to change the "warning" for just a constant WARNING, but I don't think that adding a explicit BuildWarningCheck simplifies it.
So, I'd prefer to move the literall to constants that can be changed "warning" / "warn".

Reply:
I would avoid to over-arch with this.